toolbox

```diff
@@ -748,8 +748,8 @@ enter()
     # shellcheck disable=SC2016
     # for the command passed to capsh
     $prefix_sudo podman exec \
-            $set_dbus_system_bus_address \
-            $set_environment \
+            ${set_dbus_system_bus_address:+"$set_dbus_system_bus_address"} \
+            "$set_environment" \
```

Comment:
Sadly, we can't quote these variables like this either. :(

In the case of `set_dbus_system_bus_address`, when the `DBUS_SYSTEM_BUS_ADDRESS` environment variable is present, it evaluates to:
```
podman exec ... "--env DBUS_SYSTEM_BUS_ADDRESS=..." ...
```

The entire `"--env DBUS_SYSTEM_BUS_ADDRESS=..."` string would be considered a single command line argument, instead of separately as `--env` and `DBUS_SYSTEM_BUS_ADDRESS=...`. When that happens, `podman` actually errors out.

In the case of `set_environment`, it evaluates to:
```
podman exec ... "--env=COLORTERM=truecolor --env=DBUS_SESSION_BUS_ADDRESS=..."
```

Again, the entire `"--env=COLORTERM=truecolor --env=DBUS_SESSION_BUS_ADDRESS=..."` string is considered a single command line argument, instead of multiple `--env` flags and their parameters. In this case, `podman` happens to ignore the whole string, but the spawned shell doesn't work properly because the variables aren't set. Missing `SHELL` and `TERM` seems to break Bash quite badly.

For the first case, we could use parameter expansion without the quotes:
```
${set_dbus_system_bus_address:+$set_dbus_system_bus_address}
```
It silences `SC2086` but doesn't prevent the basic problem of [globbing and word splitting](https://github.com/koalaman/shellcheck/wiki/SC2086) that it complains about.

For the second case, I don't know of any other option than to use `$set_environment` (or the identical `${set_environment}`) as before and disable the warning via a directive.